json/decode.go

```diff
@@ -617,11 +618,19 @@ func (d *decodeState) object(v reflect.Value) {
 		start := d.off - 1
 		op = d.scanWhile(scanContinue)
 		item := d.data[start : d.off-1]
-		key, ok := unquoteBytes(item)
+		key, ok := unquote(item)
```

Comment:
Sounds good to me. :+1: :shipit:
On Mar 16, 2016 8:47 AM, "Cedric Staub" <notifications@github.com> wrote:

> In json/decode.go
> <https://github.com/square/go-jose/pull/76#discussion_r56358450>:
>
> > @@ -617,11 +618,19 @@ func (d *decodeState) object(v reflect.Value) {
> >  		start := d.off - 1
> >  		op = d.scanWhile(scanContinue)
> >  		item := d.data[start : d.off-1]
> > -		key, ok := unquoteBytes(item)
> > +		key, ok := unquote(item)
>
> It was previously calling unquoteBytes so it could do case-folding (by
> calling equalFold below), but that's not relevant anymore as I deleted
> all the case-folding logic. Calling unquote is equivalent to unquoteBytes,
> except that the return value is a string instead of the raw bytes of the
> key. Changed it because maps don't allow byte slice indices (because of
> their mutability I believe).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/square/go-jose/pull/76/files/600ed83a21824967b13d7a72f5dc1591acbcb763#r56358450>
>